Dockerfile

```diff
@@ -1,34 +1,39 @@
 FROM        registry.access.redhat.com/ubi9/python-39:9.5-1739799514@sha256:9bbc4cfeac896544ab3eafa088c3d6995e82592362d374606f00d221f2986fe0 as prod
 
+COPY --from=ghcr.io/astral-sh/uv:0.6.11@sha256:fb91e82e8643382d5bce074ba0d167677d678faff4bd518dac670476d19b159c /uv /bin/uv
+
+COPY --chown=1001:0 pyproject.toml uv.lock README.md /dashdotdb/
 COPY        LICENSE /licenses/LICENSE
 USER        1001
 WORKDIR     /dashdotdb
-RUN         python3 -m venv venv
-ENV         VIRTUAL_ENV=/dashdotdb/venv
-ENV         PATH="$VIRTUAL_ENV/bin:$PATH"
+
+ENV \
+  UV_COMPILE_BYTECODE="true" \
+  UV_NO_CACHE=true
+RUN \
+  UV_DYNAMIC_VERSIONING_BYPASS="0.0.0" \
+  uv \
+    sync \
+      --frozen \
+      --python /usr/bin/python3
```

Comment:
we don't need `dev` deps in prod image, use `RUN uv sync --frozen --no-install-project --no-group dev`, checkout template https://github.com/app-sre/python-project-template/blob/main/%7B%7B%20cookiecutter.project_slug%20%7D%7D/Dockerfile